kadi/commands/tests/test_commands.py

```diff
@@ -314,6 +316,32 @@ def test_commands_create_archive_regress(tmpdir, version_env):
                 # Path("cmds_local.txt").write_text(out)
                 assert len(cmds_flight) == len(cmds_local)
 
+            # Roughly validate NSM quaternions and remove from later comparison
+            nsm_idxs = np.flatnonzero(cmds_flight['npnt_enab'] == False) # noqa
+            for idx in nsm_idxs:
+                flight_att = cmds_flight[idx]['params']['targ_att']
+                flight_q = Quat(q=flight_att)
+                local_q = Quat(q=cmds_local[idx]['params']['targ_att'])
+                delta_point = sphere_dist(flight_q.ra, flight_q.dec,
+                                          local_q.ra, local_q.dec)
+
+                assert delta_point < 0.1
+                assert np.isclose(flight_q.roll, local_q.roll, atol=0.1)
```

Comment:
I think you can do `dq = flight_q.dq(local_q)` then
```
for attr in ("roll0", "pitch", "yaw"):
    assert abs(getattr(dq, attr)) < 1e-6
```
I would expect that the arch-dependent numerical issues we are catching here should be very small. Allow for 0.1 deg seems much too large. If it is really so large then we have a different problem that should be fixed.

Reply:
I thought that for your regular kadi validation you open up to 4 degrees normal sun quaternions.  I didn't look much at the why on these numeric differences for the same kinds of quaternions as much as the don't care?  So I'm not sure about what is reasonable for tolerances.  However, if the tests aren't changing that's fine.

Reply:
The simplest thing would just be to validate _all_ the quaternions using a Quat-based check, I just wasn't as worried about simplicity for the test code.

Reply:
> I thought that for your regular kadi validation you open up to 4 degrees normal sun quaternions.

I don't know what that means. 

Fundamentally what this test is asking is: "If I run the current kadi commands update code now over a certain time range, does that match the commands that resulted from flight processing?". The only difference should be miniscule differences attributable to floating point processor differences in complex calculations. This is a sensitive test that none of the underlying code has changed in a substantive way.

If (for example) we changed `ska_sun` to provide a very slightly different sun position, then that should generate a regression failure and we decide how to handle it.

Reply:
OK.  For "I don't know what that means."  what I was trying to say was that the only quaternions in here that are "mismatched" on fido are the derived / calculated quaternions.  Those only come up in the context of NSM quaternions.  When you validate those quaternions in kadi validation against telemetry, you only require that they be within 4 degrees of telemetry.  So I had just been thinking for the specific case of these calculated quaternions, that they don't really need to be very accurate or precise, as they aren't *used* downstream in a way that actually requiest them to be accurate and precise.  But if you just want to verify that the numerics in the regression test output are not far from the test system that's fine.

Still the question then if we should just verify all the quaternions or have custom logic for the NSM quaternions.

Reply:
For all but the NSM quaternions, they are just read from the backstop, so there aren't numeric issues.